tokio-stream/src/stream_close.rs

```diff
@@ -0,0 +1,108 @@
+use crate::Stream;
+use pin_project_lite::pin_project;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+pin_project! {
+    /// Stream that returns None when the stream is exhausted. To be used with StreamMap to know when a stream ends.
+    /// # Example
+    /// Using `StreamNotifyClose` to handle closed streams with `StreamMap`
```

Comment:
The grammar here is bad. Additionally, we have empty lines before and after headings.
```suggestion
    /// A `Stream` that wraps the values in an `Option`.
    ///
    /// Whenever the wrapped stream yields an item, this stream yields that item
    /// wrapped in `Some`. When the inner stream ends, then this stream first
    /// yields a `None` item, and then this stream will also end.
    ///
    /// # Example
    ///
    /// Using `StreamNotifyClose` to handle closed streams with `StreamMap`.
```